src/django/api/serializers/v1/opensearch_common_validators/geo_polygon_validator.py

```diff
@@ -0,0 +1,68 @@
+from typing import List
+from api.serializers.v1.opensearch_validation_interface import (
+    OpenSearchValidationInterface,
+)
+from api.constants import CoordinateLimits
+
+
+class GeoPolygonValidator(OpenSearchValidationInterface):
+    def validate_opensearch_params(self, data):
+        errors: List[dict] = []
+
+        geo_polygon_data = data.get('geo_polygon', [])
+        if not isinstance(geo_polygon_data, list):
+            errors.append({
+                "field": "geo_polygon",
+                "detail": "geo_polygon must be a list of coordinates."
+            })
+            return errors
+
+        if geo_polygon_data and len(geo_polygon_data) < 3:
+            errors.append({
+                "field": "geo_polygon",
+                "detail": (
+                    "At least 3 points are required in "
+                    "geo_polygon to form a valid polygon."
+                )
+            })
+
+        for point in geo_polygon_data:
+            try:
+                lat, lon = map(float, point.split(","))
+            except (ValueError, TypeError):
+                errors.append({
+                    "field": "geo_polygon",
+                    "detail": (
+                        f"Invalid coordinate format: {point}, "
+                        "must be 'lat,lon' as floats"
```

Comment:
I believe a dot should be placed at the end of the sentence.

Reply:
Done.